plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/xtext/server/concurrent/XRequestManager.java

```diff
@@ -22,6 +22,10 @@
 import com.google.inject.Singleton;
 
 /**
+ * Handles requests from the LSP client.
```

Comment:
I don't see the synchronization here apart from the comment

Reply:
The synchronization was done already before in commit 8e74fc218c6da26e56288fe271edce2aed063478